### ISSUE TITLE
Integer overflow in elm SpaceAge test

### ISSUE THREAD

**Issue by Phippsaurus** [exercism/exercism.io#3698]
Elm currently has a [problem with big integers](https://github.com/elm-lang/elm-compiler/issues/1648).
This causes the _SpaceAge_ exercise to fail half the tests with a correct implementation, as the integer literals are wrapped into the negative.

Therefore e.g. 

`````elm
test "age on neptune" <|
                \() -> Expect.equal 2 (round (ageOn Neptune 8210123456))
`````
really passes `-379811136` to the `ageOn` function.

**Comment by NobbZ** [comment]
I can't remember having this problem when o solved the exercise, but I didn't use integers...

**Comment by Phippsaurus** [comment]
I have just checked, and this problem only exists on WIndows 10. I tried on Manjaro Linux and everything works as expected.

My method signature is

`````elm
ageOn : Planet -> Float -> Float
`````
, so I do not directly use integers either.
If I change the number literals in the tests to e.g.

`````elm
test "age on neptune" <|
    \() -> Expect.equal 2 (round (ageOn Neptune 8210123456.0))
`````
it works like a charm on Windows too.

The problem seems to be that the literal `8210123456` is parsed as an integer before type inference takes place, and then cast to `Float`, once its type is deduced from the signature of the function it is passed into.

In `elm-repl` I get

`````elm-repl
> 8210123456
-379811136 : number
> 8210123456.0
8210123456 : Float
> sqrt <| 8210123456^2
379811136 : Float
`````
so I believe this is a more general problem with elm on Windows. On Linux the repl output also looks as expected.
